compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -479,11 +479,11 @@ object ProtoTypes {
    *  for each parameter.
    *  @return  The added type lambda, and the list of created type variables.
    */
-  def constrained(tl: TypeLambda, owningTree: untpd.Tree, alwaysAddTypeVars: Boolean = false)(implicit ctx: Context): (TypeLambda, List[TypeTree]) = {
+  def constrained(tl: TypeLambda, owningTree: untpd.Tree, alwaysAddTypeVars: Boolean)(implicit ctx: Context): (TypeLambda, List[TypeTree]) = {
     val state = ctx.typerState
     val addTypeVars = alwaysAddTypeVars || !owningTree.isEmpty
     if (tl.isInstanceOf[PolyType])
-      assert(!(ctx.typerState.isCommittable && !addTypeVars),
+      assert(!ctx.typerState.isCommittable || addTypeVars,
         s"inconsistent: no typevars were added to committable constraint ${state.constraint}")
```

Comment:
If we change the assertion then this message also needs to be updated I think:
```suggestion
        s"inconsistent: typevars were added to non-committable constraint ${state.constraint}")
```

Reply:
ah nevermind I got confused, the current message is correct.